File: app/src/main/java/io/github/qwqgong/androidcyaml/ProcessMatchingMode.java
package io.github.qwqgong.androidcyaml;

import java.util.Locale;

enum ProcessMatchingMode {
    STRICT("strict"),
    ALWAYS("always"),
    OFF("off");

    private final String wireValue;

    ProcessMatchingMode(String wireValue) {
        this.wireValue = wireValue;
    }

    String wireValue() {
        return wireValue;
    }

    static ProcessMatchingMode fromWireValue(String value) {
        String normalized = value == null
                ? ""
                : value.trim().toLowerCase(Locale.ROOT);
        if (normalized.isEmpty()) {
            return ALWAYS;
        }
        for (ProcessMatchingMode mode : values()) {
            if (mode.wireValue.equals(normalized)) {
                return mode;
            }
        }
        throw new IllegalArgumentException("不支持的进程匹配模式：" + value);
    }
}

File: app/src/main/java/io/github/qwqgong/androidcyaml/ProcessMatchingMode.kt
package io.github.qwqgong.androidcyaml

import java.util.Locale

enum class ProcessMatchingMode(val wireValue: String) {
    STRICT("strict"),
    ALWAYS("always"),
    OFF("off");

    companion object {
        fun fromWireValue(value: String?): ProcessMatchingMode {
            val normalized = value?.trim()?.lowercase(Locale.ROOT) ?: ""
            if (normalized.isEmpty()) {
                return ALWAYS
            }
            for (mode in values()) {
                if (mode.wireValue == normalized) {
                    return mode
                }
            }
            throw IllegalArgumentException("不支持的进程匹配模式：$value")
        }
    }
}
